tools/nd_freeze

```diff
@@ -0,0 +1,334 @@
+#!/usr/bin/perl
+#
+# Install access to the NeuroDebian snapshot repository that allows access
+# to old packages based on dates and version numbers.
+#
+# The UTC date/time format sent to NeuroDebian to get the repo info is in the
+# format yyyymmddThhmmssz or yyymmdd. However, the script will handle and
+# reformat a number of date/time formats. Quotes are required if there is a
+# space between the date and time components:
+#
+#   yyyymmddThh:mm:ssZ
+#   yyyymmdd
+#   mm/dd/yyyy
+#   yyyy-mm-dd
+#   yyyymmddThh:mm
+#   "yyyy-mm-dd hh:mm:ss"
+#   "mm/dd/yy hh:mm:ss"
+#
+# Browse available snapshots here:
+# http://snapshot-neuro.debian.net:5002/archive/neurodebian/
+#
+# If there is no import at the exact time you specified you will get both the
+# previous and next available timestamped repos of the one you specified to make
+# available versions that may have been placed in the repo before your
+# selected time but after the previous snapshot.
+
+require 5.002;
+use Socket;
+
+
+# Function returns the page contents of the provided URL. We use the Sockets
+# library because it has a high probability of being provided in the perl
+# installation for the environment and don't have to download any packages.
+# 
+# Parameters
+# ----------
+# $url
+#     string : URL of web page of interest
+# 
+# Returns
+# -------
+# string : HTML page contents of URL
+# 
+sub get_www_content {
+    my ($url) = @_;
+    $url =~ /http:\/\/([^\/:]+):?(\d+)*(.*)/;
+    my $dest = $1;
+    my $port = $2;
+    my $file = $3;
+    $port = 80 if (!$port);
+    my $proto = getprotobyname('tcp');
+    socket(F, PF_INET, SOCK_STREAM, $proto);
+    my $sin = sockaddr_in($port,inet_aton($dest));
+    connect(F, $sin) || return undef;
+    my $old_fh = select(F);
+    $| = 1;
+    select($old_fh);
+    print F "GET $file HTTP/1.1\nHost: ${dest}\n\n";
+    $/ = undef;
+    $contents = <F>;
+    close(F);
+    return $contents;
+}
+
+
+# Function to run apt-get update
+sub run_apt_get_update {
+    print "Refreshing apt cache ...\n";
+    qx!apt-get --allow-original-apt-sources update 2> /dev/null!;
```

Comment:
where did you find this option for `apt-get`?  or is that supposed to be an option for this tool?

Reply:
and of cause with `2> /dev/null` no error is spit out ;)  better not to hide those that well!